File: Tests/Types/AnnotationsNullable/SelfReferencingType.cs
using KY.Generator;

namespace Types;

[GenerateTypeScriptModel]
[GenerateStrict]
public class SelfReferencingType
{
    public string StringProperty { get; set; } = "";
    public SelfReferencingType? SelfProperty { get; set; }
    public List<SelfReferencingType> SelfList { get; set; } = [];
}

File: Tests/v10/SelfReferencingAnnotationsNullable/Output/self-referencing-type.ts
/* eslint-disable */
// tslint:disable

export class SelfReferencingType {
    public stringProperty: string = "";
    public selfProperty?: SelfReferencingType;
    public selfList: SelfReferencingType[] = [];

    public constructor(init?: Partial<SelfReferencingType>) {
        Object.assign(this, init);
    }
}

// outputid:b84d2f96-fba8-416c-b067-f17f09809015
